lib/core/application/vault/deploy_vault_with_owner_key_bloc/deploy_vault_with_owner_key_bloc.dart

```diff
@@ -0,0 +1,155 @@
+import 'dart:async';
+
+import 'package:app/core/domain/web3/entities/chain.dart';
+import 'package:app/core/domain/web3/entities/raw_transaction.dart';
+import 'package:app/core/service/vault/private_key/private_key.dart';
+import 'package:flutter/material.dart';
+import 'package:flutter_bloc/flutter_bloc.dart';
+import 'package:freezed_annotation/freezed_annotation.dart';
+import 'package:web3dart/crypto.dart';
+import 'package:web3dart/json_rpc.dart';
+import 'package:web3dart/web3dart.dart';
+import 'package:http/http.dart' as http;
+
+part 'deploy_vault_with_owner_key_bloc.freezed.dart';
+
+class DeployVaultWithOwnerKeyBloc
+    extends Bloc<DeployVaultWithOwnerKeyEvent, DeployVaultWithOwnerKeyState> {
+  StreamSubscription? _subscription;
+  DeployVaultWithOwnerKeyBloc() : super(DeployVaultWithOwnerKeyState.idle()) {
+    on<DeployVaultWithOwnerKeyEventStartDeploy>(_onStartDeploy);
+  }
+  int maxAttempt = 20;
+  int attempt = 0;
+
+  @override
+  Future<void> close() async {
+    await _subscription?.cancel();
+    super.close();
+  }
+
+  Future<void> _onStartDeploy(
+    DeployVaultWithOwnerKeyEventStartDeploy event,
+    Emitter emit,
+  ) async {
+    try {
+      emit(
+        DeployVaultWithOwnerKeyState.loading(),
+      );
+      final network = event.network;
+      final privateKey = await PrivateKey.get(
+        PrivateKey.identifier(
+          EthereumAddress.fromHex(event.ownerAddress),
+        ),
+      );
+      if (privateKey == null) {
```

Comment:
If privateKey == null is it causing infinity loading ? because we return here instead of emit failure

Reply:
yes, let's me fix that